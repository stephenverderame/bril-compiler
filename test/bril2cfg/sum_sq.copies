digraph {
	subgraph cluster_sumOfSquares {
		label="sumOfSquares(n: int)";
		rankdir="TB";
		sumOfSquares_0 [label="START", shape="rectangle", style="rounded"];
		sumOfSquares_2 [label="[]\n----------\nv0 := 0\nres := v0\nv2 := 1\ni := v2\n----------\n[i = v2, res = v0]", shape="rectangle", style="rounded"];
		sumOfSquares_3 [label="[sq = v8, v10 = sq, v4 = n]\n----------\nv3 := i\nv4 := n\nv5 := v3 <= v4\nif v5\n----------\n[sq = v8, v10 = sq, v3 = i, v4 = n]", shape="rectangle", style="rounded"];
		sumOfSquares_4 [label="[sq = v8, v10 = sq, v3 = i, v4 = n]\n----------\nv6 := i\nv7 := i\nv8 := v6 * v7\nsq := v8\nv9 := res\nv10 := sq\nv11 := v9 + v10\nres := v11\nv12 := i\nv13 := 1\nv14 := v12 + v13\ni := v14\n----------\n[i = v14, res = v11, sq = v8, v10 = sq, v4 = n]", shape="rectangle", style="rounded"];
		sumOfSquares_5 [label="[sq = v8, v10 = sq, v3 = i, v4 = n]\n----------\nv15 := res\nreturn v15\n----------\n[sq = v8, v10 = sq, v15 = res, v3 = i, v4 = n]", shape="rectangle", style="rounded"];

		sumOfSquares_0 -> sumOfSquares_2;
		sumOfSquares_2 -> sumOfSquares_3;
		sumOfSquares_3 -> sumOfSquares_4 [label="T"];
		sumOfSquares_3 -> sumOfSquares_5 [label="F"];
		sumOfSquares_4 -> sumOfSquares_3;
	}
	subgraph cluster_squareOfSum {
		label="squareOfSum(n: int)";
		rankdir="TB";
		squareOfSum_0 [label="START", shape="rectangle", style="rounded"];
		squareOfSum_2 [label="[]\n----------\nv0 := 0\nres := v0\nv2 := 1\ni := v2\n----------\n[i = v2, res = v0]", shape="rectangle", style="rounded"];
		squareOfSum_3 [label="[v4 = n]\n----------\nv3 := i\nv4 := n\nv5 := v3 <= v4\nif v5\n----------\n[v3 = i, v4 = n]", shape="rectangle", style="rounded"];
		squareOfSum_4 [label="[v3 = i, v4 = n]\n----------\nv6 := res\nv7 := i\nv8 := v6 + v7\nres := v8\nv9 := i\nv10 := 1\nv11 := v9 + v10\ni := v11\n----------\n[i = v11, res = v8, v4 = n]", shape="rectangle", style="rounded"];
		squareOfSum_5 [label="[v3 = i, v4 = n]\n----------\nv12 := res\nv13 := res\nv14 := v12 * v13\nsquare := v14\nv15 := square\nreturn v15\n----------\n[square = v14, v12 = res, v13 = res, v15 = square, v3 = i, v4 = n]", shape="rectangle", style="rounded"];

		squareOfSum_0 -> squareOfSum_2;
		squareOfSum_2 -> squareOfSum_3;
		squareOfSum_3 -> squareOfSum_4 [label="T"];
		squareOfSum_3 -> squareOfSum_5 [label="F"];
		squareOfSum_4 -> squareOfSum_3;
	}
	subgraph cluster_main {
		label="main(n: int)";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="[]\n----------\nv0 := n\nsum := sumOfSquares(v0)\nsum := sum\nv1 := n\nsquare := squareOfSum(v1)\nsquare := square\nv2 := square\nv3 := sum\nv4 := v2 - v3\ndiff := v4\nv5 := diff\nprint v5\nv6 := 0\nv7 := diff\n----------\n[diff = v4, square = square, sum = sum, v0 = n, v1 = n, v2 = square, v3 = sum, v5 = diff, v7 = diff]", shape="rectangle", style="rounded"];

		main_0 -> main_2;
	}
}
