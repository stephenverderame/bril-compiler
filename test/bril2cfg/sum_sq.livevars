digraph {
	subgraph cluster_sumOfSquares {
		label="sumOfSquares(n: int)";
		rankdir="TB";
		sumOfSquares_0 [label="START", shape="rectangle", style="rounded"];
		sumOfSquares_2 [label="[n]\n----------\nv0 := 0\nres := v0\nv2 := 1\ni := v2\n----------\n[i, n, res]", shape="rectangle", style="rounded"];
		sumOfSquares_3 [label="[i, n, res]\n----------\nv3 := i\nv4 := n\nv5 := v3 <= v4\nif v5\n----------\n[i, n, res]", shape="rectangle", style="rounded"];
		sumOfSquares_4 [label="[i, n, res]\n----------\nv6 := i\nv7 := i\nv8 := v6 * v7\nsq := v8\nv9 := res\nv10 := sq\nv11 := v9 + v10\nres := v11\nv12 := i\nv13 := 1\nv14 := v12 + v13\ni := v14\n----------\n[i, n, res]", shape="rectangle", style="rounded"];
		sumOfSquares_5 [label="[res]\n----------\nv15 := res\nreturn v15\n----------\n[]", shape="rectangle", style="rounded"];

		sumOfSquares_0 -> sumOfSquares_2;
		sumOfSquares_2 -> sumOfSquares_3;
		sumOfSquares_3 -> sumOfSquares_4 [label="T"];
		sumOfSquares_3 -> sumOfSquares_5 [label="F"];
		sumOfSquares_4 -> sumOfSquares_3;
	}
	subgraph cluster_squareOfSum {
		label="squareOfSum(n: int)";
		rankdir="TB";
		squareOfSum_0 [label="START", shape="rectangle", style="rounded"];
		squareOfSum_2 [label="[n]\n----------\nv0 := 0\nres := v0\nv2 := 1\ni := v2\n----------\n[i, n, res]", shape="rectangle", style="rounded"];
		squareOfSum_3 [label="[i, n, res]\n----------\nv3 := i\nv4 := n\nv5 := v3 <= v4\nif v5\n----------\n[i, n, res]", shape="rectangle", style="rounded"];
		squareOfSum_4 [label="[i, n, res]\n----------\nv6 := res\nv7 := i\nv8 := v6 + v7\nres := v8\nv9 := i\nv10 := 1\nv11 := v9 + v10\ni := v11\n----------\n[i, n, res]", shape="rectangle", style="rounded"];
		squareOfSum_5 [label="[res]\n----------\nv12 := res\nv13 := res\nv14 := v12 * v13\nsquare := v14\nv15 := square\nreturn v15\n----------\n[]", shape="rectangle", style="rounded"];

		squareOfSum_0 -> squareOfSum_2;
		squareOfSum_2 -> squareOfSum_3;
		squareOfSum_3 -> squareOfSum_4 [label="T"];
		squareOfSum_3 -> squareOfSum_5 [label="F"];
		squareOfSum_4 -> squareOfSum_3;
	}
	subgraph cluster_main {
		label="main(n: int)";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="[n]\n----------\nv0 := n\nsum := sumOfSquares(v0)\nsum := sum\nv1 := n\nsquare := squareOfSum(v1)\nsquare := square\nv2 := square\nv3 := sum\nv4 := v2 - v3\ndiff := v4\nv5 := diff\nprint v5\nv6 := 0\nv7 := diff\n----------\n[]", shape="rectangle", style="rounded"];

		main_0 -> main_2;
	}
}
