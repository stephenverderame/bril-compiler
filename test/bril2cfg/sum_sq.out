digraph {
	subgraph cluster_sumOfSquares {
		label="sumOfSquares";
		rankdir="TB";
		sumOfSquares_0 [label="START", shape="ellipse"];
		sumOfSquares_2 [label="v0 := 0", shape="ellipse"];
		sumOfSquares_3 [label="res := v0", shape="ellipse"];
		sumOfSquares_4 [label="v2 := 1", shape="ellipse"];
		sumOfSquares_5 [label="i := v2", shape="ellipse"];
		sumOfSquares_6 [label="v3 := i", shape="ellipse"];
		sumOfSquares_7 [label="v4 := n", shape="ellipse"];
		sumOfSquares_8 [label="v5 := v3 <= v4", shape="ellipse"];
		sumOfSquares_9 [label="if v5", shape="ellipse"];
		sumOfSquares_10 [label="v6 := i", shape="ellipse"];
		sumOfSquares_11 [label="v7 := i", shape="ellipse"];
		sumOfSquares_12 [label="v8 := v6 * v7", shape="ellipse"];
		sumOfSquares_13 [label="sq := v8", shape="ellipse"];
		sumOfSquares_14 [label="v9 := res", shape="ellipse"];
		sumOfSquares_15 [label="v10 := sq", shape="ellipse"];
		sumOfSquares_16 [label="v11 := v9 + v10", shape="ellipse"];
		sumOfSquares_17 [label="res := v11", shape="ellipse"];
		sumOfSquares_18 [label="v12 := i", shape="ellipse"];
		sumOfSquares_19 [label="v13 := 1", shape="ellipse"];
		sumOfSquares_20 [label="v14 := v12 + v13", shape="ellipse"];
		sumOfSquares_21 [label="i := v14", shape="ellipse"];
		sumOfSquares_23 [label="v15 := res", shape="ellipse"];
		sumOfSquares_24 [label="return v15", shape="ellipse"];

		sumOfSquares_0 -> sumOfSquares_2;
		sumOfSquares_2 -> sumOfSquares_3;
		sumOfSquares_3 -> sumOfSquares_4;
		sumOfSquares_4 -> sumOfSquares_5;
		sumOfSquares_5 -> sumOfSquares_6;
		sumOfSquares_6 -> sumOfSquares_7;
		sumOfSquares_7 -> sumOfSquares_8;
		sumOfSquares_8 -> sumOfSquares_9;
		sumOfSquares_9 -> sumOfSquares_10 [label="T"];
		sumOfSquares_9 -> sumOfSquares_23 [label="F"];
		sumOfSquares_10 -> sumOfSquares_11;
		sumOfSquares_11 -> sumOfSquares_12;
		sumOfSquares_12 -> sumOfSquares_13;
		sumOfSquares_13 -> sumOfSquares_14;
		sumOfSquares_14 -> sumOfSquares_15;
		sumOfSquares_15 -> sumOfSquares_16;
		sumOfSquares_16 -> sumOfSquares_17;
		sumOfSquares_17 -> sumOfSquares_18;
		sumOfSquares_18 -> sumOfSquares_19;
		sumOfSquares_19 -> sumOfSquares_20;
		sumOfSquares_20 -> sumOfSquares_21;
		sumOfSquares_21 -> sumOfSquares_6;
		sumOfSquares_23 -> sumOfSquares_24;
	}
	subgraph cluster_squareOfSum {
		label="squareOfSum";
		rankdir="TB";
		squareOfSum_0 [label="START", shape="ellipse"];
		squareOfSum_2 [label="v0 := 0", shape="ellipse"];
		squareOfSum_3 [label="res := v0", shape="ellipse"];
		squareOfSum_4 [label="v2 := 1", shape="ellipse"];
		squareOfSum_5 [label="i := v2", shape="ellipse"];
		squareOfSum_6 [label="v3 := i", shape="ellipse"];
		squareOfSum_7 [label="v4 := n", shape="ellipse"];
		squareOfSum_8 [label="v5 := v3 <= v4", shape="ellipse"];
		squareOfSum_9 [label="if v5", shape="ellipse"];
		squareOfSum_10 [label="v6 := res", shape="ellipse"];
		squareOfSum_11 [label="v7 := i", shape="ellipse"];
		squareOfSum_12 [label="v8 := v6 + v7", shape="ellipse"];
		squareOfSum_13 [label="res := v8", shape="ellipse"];
		squareOfSum_14 [label="v9 := i", shape="ellipse"];
		squareOfSum_15 [label="v10 := 1", shape="ellipse"];
		squareOfSum_16 [label="v11 := v9 + v10", shape="ellipse"];
		squareOfSum_17 [label="i := v11", shape="ellipse"];
		squareOfSum_19 [label="v12 := res", shape="ellipse"];
		squareOfSum_20 [label="v13 := res", shape="ellipse"];
		squareOfSum_21 [label="v14 := v12 * v13", shape="ellipse"];
		squareOfSum_22 [label="square := v14", shape="ellipse"];
		squareOfSum_23 [label="v15 := square", shape="ellipse"];
		squareOfSum_24 [label="return v15", shape="ellipse"];

		squareOfSum_0 -> squareOfSum_2;
		squareOfSum_2 -> squareOfSum_3;
		squareOfSum_3 -> squareOfSum_4;
		squareOfSum_4 -> squareOfSum_5;
		squareOfSum_5 -> squareOfSum_6;
		squareOfSum_6 -> squareOfSum_7;
		squareOfSum_7 -> squareOfSum_8;
		squareOfSum_8 -> squareOfSum_9;
		squareOfSum_9 -> squareOfSum_10 [label="T"];
		squareOfSum_9 -> squareOfSum_19 [label="F"];
		squareOfSum_10 -> squareOfSum_11;
		squareOfSum_11 -> squareOfSum_12;
		squareOfSum_12 -> squareOfSum_13;
		squareOfSum_13 -> squareOfSum_14;
		squareOfSum_14 -> squareOfSum_15;
		squareOfSum_15 -> squareOfSum_16;
		squareOfSum_16 -> squareOfSum_17;
		squareOfSum_17 -> squareOfSum_6;
		squareOfSum_19 -> squareOfSum_20;
		squareOfSum_20 -> squareOfSum_21;
		squareOfSum_21 -> squareOfSum_22;
		squareOfSum_22 -> squareOfSum_23;
		squareOfSum_23 -> squareOfSum_24;
	}
	subgraph cluster_main {
		label="main";
		rankdir="TB";
		main_0 [label="START", shape="ellipse"];
		main_2 [label="v0 := n", shape="ellipse"];
		main_3 [label="sum := sumOfSquares(v0)", shape="ellipse"];
		main_4 [label="sum := sum", shape="ellipse"];
		main_5 [label="v1 := n", shape="ellipse"];
		main_6 [label="square := squareOfSum(v1)", shape="ellipse"];
		main_7 [label="square := square", shape="ellipse"];
		main_8 [label="v2 := square", shape="ellipse"];
		main_9 [label="v3 := sum", shape="ellipse"];
		main_10 [label="v4 := v2 - v3", shape="ellipse"];
		main_11 [label="diff := v4", shape="ellipse"];
		main_12 [label="v5 := diff", shape="ellipse"];
		main_13 [label="print v5", shape="ellipse"];
		main_14 [label="v6 := 0", shape="ellipse"];
		main_15 [label="v7 := diff", shape="ellipse"];

		main_0 -> main_2;
		main_2 -> main_3;
		main_3 -> main_4;
		main_4 -> main_5;
		main_5 -> main_6;
		main_6 -> main_7;
		main_7 -> main_8;
		main_8 -> main_9;
		main_9 -> main_10;
		main_10 -> main_11;
		main_11 -> main_12;
		main_12 -> main_13;
		main_13 -> main_14;
		main_14 -> main_15;
	}
}
